lib/node_modules/@stdlib/blas/base/dtrsm/README.md

```diff
@@ -0,0 +1,247 @@
+
+<section class = "usage">
+
+## Usage
+
+```javascript
+var dtrsm = require( '@stdlib/blas/base/dtrsm' );
+```
+
+#### dtrsm( order, side, uplo, transa, diag, m, n, alpha, A, LDA, B, LDB )
+
+Solves matrix equation `op(A) * X = alpha * B` or `X * op(A) = alpha * B` where `alpha` is a scalar, `X` and `B` are m by n matrices, `A` is a unit, or non-unit, upper or lower triangular matrix and `op( A )` is one of `op( A ) = A` or `op( A ) = A ** T`. The matrix `X` is overwritten on `B`.
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+A = new Float64Array( [ 1.0, 3.0, 0.0, 4.0 ] );
+B = new Float64Array( [ 5.0, 7.0, 0.0, 8.0 ] );
+
+dtrsm( 'row-major', 'left', 'upper', 'none', 'non-unit', 2, 2, 6.0, A, 2, B, 2 );
+// B => <Float64Array>[ 30.0, 6.0, 0.0, 12.0 ]
+```
+
+The function has the following parameters:
+
+-   **order**: storage layout of `A` and `B`.
+-   **side**: specifies whether `op( A )` appears on the left or right side of `X`.
+-   **uplo**: specifies whether the upper or lower triangular part of the matrix `A` is supplied.
+-   **transa**: specifies the form of `op( A )` to be used in the matrix multiplication.
+-   **diag**: specifies whether or not `A` is unit triangular.
+-   **m**: number of rows in `B`.
+-   **n**: number of columns in `B`.
+-   **alpha**: scalar constant.
+-   **A**: input matrix `A`.
+-   **LDA**: stride of the first dimension of `A` (a.k.a., leading dimension of the matrix `A`).
+-   **B**: input matrix `B`.
+-   **LDB**: stride of the first dimension of `B` (a.k.a., leading dimension of the matrix `B`).
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+A = new Float64Array( [ 1.0, 3.0, 0.0, 4.0 ] );
+B = new Float64Array( [ 5.0, 7.0, 0.0, 8.0 ] );
+
+dtrsm( 'row-major', 'left', 'upper', 'none', 'non-unit', 2, 2, 6.0, A, 2, B, 2 );
+// B => <Float64Array>[ 30.0, 6.0, 0.0, 12.0 ]
+```
+
+Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
+
+<!-- eslint-disable stdlib/capitalized-comments -->
+
+```javascript
+var Float64Array = require( '@stdlib/array/float64' );
+
+// Initial arrays...
+var A0 = new Float64Array( [ 0.0, 1.0, 3.0, 0.0, 4.0 ] );
+var B0 = new Float64Array( [ 0.0, 5.0, 7.0, 0.0, 8.0 ] );
+
+// Create offset views...
+var A1 = new Float64Array( A0.buffer, A0.BYTES_PER_ELEMENT*1 ); // start at 1st element
+var B1 = new Float64Array( B0.buffer, B0.BYTES_PER_ELEMENT*1 ); // start at 1st element
+
+dtrsm( 'row-major', 'left', 'upper', 'none', 'non-unit', 2, 2, 6.0, A1, 2, B1, 2 );
+// B0 => <Float64Array>[ 0.0, 30.0, 6.0, 0.0, 12.0 ]
+```
+
+#### dtrsm.ndarray( ord, side, ul, ta, diag, m, n, α, A, LDA, oa, B, LDB, ob )
```

Comment:
@Pranavchiku Would you mind updating this implementation and associated docs and tests to support separate dimension strides for `A` and `B`?